test/post_processing/test_roc_curve_based_post_processing.py

```diff
@@ -30,150 +31,163 @@
 
-    with pytest.raises(ValueError) as exception:
-        adjusted_model.fit(X, Y, A)
-    assert str(exception.value) == error_message
+    if X is None or Y is None and A is None:
+        with pytest.raises(ValueError) as exception:
+            adjusted_model.fit(X, Y, A)
+        assert str(exception.value) == error_message
 
 
-@pytest.mark.parametrize("formatting_function", [lambda x: x, np.array])
+@pytest.mark.parametrize("X_transform", ALLOWED_INPUT_DATA_TYPES)
+@pytest.mark.parametrize("Y_transform", ALLOWED_INPUT_DATA_TYPES)
+@pytest.mark.parametrize("A_transform", ALLOWED_INPUT_DATA_TYPES)
 @pytest.mark.parametrize('Metric', [DemographicParity, EqualizedOdds])
-def test_roc_curve_based_post_processing_non_binary_labels(formatting_function, Metric):
+def test_roc_curve_based_post_processing_non_binary_labels(X_transform, Y_transform, A_transform,
+                                                           Metric):
     non_binary_labels = copy.deepcopy(example_labels)
     non_binary_labels[0] = 2
 
-    X, Y, A = _format_X_Y_A(formatting_function, _format_as_list_of_lists(example_attributes1),
-                            non_binary_labels, example_attributes1)
+    X = X_transform(_format_as_list_of_lists(example_attributes1))
+    Y = Y_transform(non_binary_labels)
+    A = A_transform(example_attributes1)
+
     adjusted_model = ROCCurveBasedPostProcessing(fairness_unaware_model=ExampleModel(),
                                                  fairness_metric=Metric())
 
     with pytest.raises(ValueError, match=NON_BINARY_LABELS_ERROR_MESSAGE):
         adjusted_model.fit(X, Y, A)
 
 
-@pytest.mark.parametrize("formatting_function", [lambda x: x, np.array])
+@pytest.mark.parametrize("X_transform", ALLOWED_INPUT_DATA_TYPES)
+@pytest.mark.parametrize("Y_transform", ALLOWED_INPUT_DATA_TYPES)
+@pytest.mark.parametrize("A_transform", ALLOWED_INPUT_DATA_TYPES)
 @pytest.mark.parametrize('Metric', [DemographicParity, EqualizedOdds])
-def test_roc_curve_based_post_processing_different_input_lengths(formatting_function, Metric):
+def test_roc_curve_based_post_processing_different_input_lengths(X_transform, Y_transform,
+                                                                 A_transform, Metric):
     n = len(example_attributes1)
     for permutation in [(0, 1), (1, 0)]:
-        with pytest.raises(ValueError, match=DIFFERENT_INPUT_LENGTH_ERROR_MESSAGE):
-            X, Y, A = _format_X_Y_A(formatting_function,
-                                    _format_as_list_of_lists(example_attributes1)[:n-permutation[0]],  # noqa: E501
-                                    example_labels[:n-permutation[1]],
-                                    example_attributes1)
+        with pytest.raises(ValueError, match=DIFFERENT_INPUT_LENGTH_ERROR_MESSAGE
+                           .format("X, aux_data, and y")):
+            X = X_transform(_format_as_list_of_lists(example_attributes1)[:n-permutation[0]])
```

Comment:
If these transforms aren't what are under test, should they go outside the `with pytest.raises` section?

Reply:
Right! I had this in a different structure before, and clearly didn't pay attention when restructuring it.

Reply:
actually these need to be here because we're changing the data every time